cmd/provisioner-localpv/app/helper_blockdevice.go

```diff
@@ -120,14 +124,20 @@ func (p *Provisioner) createBlockDeviceClaim(blkDevOpts *HelperBlockDeviceOption
 		return nil
 	}
 
-	bdcObj, err := blockdeviceclaim.NewBuilder().
+	bdcObjBuilder := blockdeviceclaim.NewBuilder().
 		WithNamespace(p.namespace).
 		WithName(bdcName).
 		WithHostName(blkDevOpts.nodeHostname).
 		WithCapacity(blkDevOpts.capacity).
 		WithFinalizer(LocalPVFinalizer).
-		WithBlockVolumeMode(blkDevOpts.volumeMode).
-		Build()
+		WithBlockVolumeMode(blkDevOpts.volumeMode)
+
+	// If bdPoolName is configure, set it on the BDC
+	if len(blkDevOpts.bdPoolName) > 0 {
```

Comment:
should we make poolName as compulsory? This will avoid for any race between app/pvc deployment and labeling of BDs.
Even though this doesn't happen usually, this can happen with south bound provisioner.